staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go

```diff
@@ -219,3 +220,50 @@ func TestNeedsHealthCheck(t *testing.T) {
 		},
 	})
 }
+
+func TestHasFinalizer(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		svc          *v1.Service
+		hasFinalizer bool
+	}{
+		{
+			desc:         "service without finalizer",
+			svc:          &v1.Service{},
+			hasFinalizer: false,
+		},
+		{
+			desc: "service with unrelated finalizer",
+			svc: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Finalizers: []string{"unrelated"},
+				},
+			},
+			hasFinalizer: false,
+		},
+		{
+			desc: "service with one finalizer",
+			svc: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Finalizers: []string{LoadBalancerCleanupFinalizer},
+				},
+			},
+			hasFinalizer: true,
+		},
+		{
+			desc: "service with multiple finalizers",
+			svc: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Finalizers: []string{LoadBalancerCleanupFinalizer, "unrelated"},
+				},
+			},
+			hasFinalizer: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		if hasFinalizer := HasFinalizer(tc.svc); hasFinalizer != tc.hasFinalizer {
+			t.Errorf("%s: HasFinalizer() = %t, want %t", tc.desc, hasFinalizer, tc.hasFinalizer)
```

Comment:
If you run this in `t.Run(tc.desc, func(t *testing) { ... })` you don't have to include the test description in the error.

Reply:
Acked, will update in the next push.